cirq-core/cirq/ops/raw_types.py

```diff
@@ -777,6 +788,12 @@ def _equal_up_to_global_phase_(
     ) -> Union[NotImplementedType, bool]:
         return protocols.equal_up_to_global_phase(self.sub_operation, other, atol=atol)
 
+    def unconditionally(self) -> Operation:
+        return TaggedOperation(self.sub_operation.unconditionally(), *self.tags)
```

Comment:
This isn't right. The tags should be removed, *unless* `self.sub_operation is self.sub_operation.unconditionally()`. Basically if it leaves the suboperation unchanged, return self, otherwise don't retag the returned operation.